Remotion/Web/Development.WebTesting/CoypuElementScopeExtensions.cs

```diff
@@ -248,5 +248,25 @@ public static void Unhover ([NotNull] this ElementScope scope)
 
       Cursor.Position = new Point (0, 0);
     }
+
+    /// <summary>
+    /// Gets an attribute from the scope or throws a <see cref="MissingHtmlException"/> if it could not be found. Wraps <see cref="ElementScope.this"/>.
+    /// </summary>
+    /// <remarks>
+    /// Accessing HTML attributes through <see cref="ElementScope.this"/> returns null if the attribute does not exist.
+    /// This might not be the desired behavior in some cases, thus offering this extension method.
+    /// </remarks>
+    /// <exception cref="MissingHtmlException">If the attribute cannot be found.</exception>
+    [NotNull]
+    public static string GetAttributeOrThrow ([NotNull] this ElementScope scope, [NotNull] string attributeName)

```

Comment:
What is the reasoning for having an `OrThrow` suffix in the method name? I was under the impression that the naming pattern was "normal" names for methods that throw exceptions and `OrDefault` or `OrNull` suffixes for methods that don't.

Reply:
That's something to check with MK.

Reply:
I like the Name, as there is this confusion in the Coypu API which operations can throw and which not (for me at least).

Reply:
I don't :)

The annotation and documentation should be sufficient. There's also no immediately obvious null-returning API on ElementScope so this should be okay. *OrThrow is just weird in most code. Also, R# didn't find a single occurance of "*OrThrow" in re-motion or Coypu, Selenium.

Reply:
Fixed in https://github.com/re-motion/Trunk-SVN2Git/pull/11/commits/9f3d7bc46ab13a1cc811bed081e65596ccc91a3c.